libexec/rtld-elf/aarch64/rtld_c18n_asm.S

```diff
@@ -30,102 +30,26 @@
 #include "rtld_c18n_machdep.h"
 #undef IN_ASM
 
-ENTRY(_rtld_thread_start)
-	mov	c1, csp
-	sub	csp, csp, #(CAP_WIDTH * 2)
-	/*
-	 * This function MUST preserve the value of c0.
-	 */
-	bl	c18n_init_rtld_stack
-#ifndef __ARM_MORELLO_PURECAP_BENCHMARK_ABI
-	/*
-	 * Move the Executive mode thread pointer to Restricted mode.
-	 */
-	mrs	c10, ctpidr_el0
-	msr	rctpidr_el0, c10
-#endif
-	b	_rtld_thread_start_impl
-END(_rtld_thread_start)
-
 /*
- * void _rtld_sighandler(int, siginfo_t *, void *);
+ * Overview
```

Comment:
This is an overview of each function and not an overview of the design and how the various parts fit together.  What we need to understand the overall code is the intention of code, how the high level design works, what happens to things like the stack, registers etc.  Each of the comments below is more appropriate to each function and they should be moved nearer to those.